src/hypertable.c

```diff
@@ -2551,6 +2552,30 @@ ts_hypertable_create_compressed(Oid table_relid, int32 hypertable_id)
 	Relation rel;
 
 	rel = table_open(table_relid, AccessExclusiveLock);
+	int32 row_size = MAXALIGN(SizeofHeapTupleHeader);
+	/* estimate tuple width of compressed hypertable */
+	row_size += get_relation_data_width(table_relid, NULL);
+	/* assuming data gets compressed to half of its original size */
+	if ((row_size / 2) > MaxHeapTupleSize)
+	{
+#ifndef NDEBUG
+		ereport(WARNING,
+				(errmsg("Estimated row size of compressed hypertable is %u. Assuming data gets "
+						"compressed to half the original size which is %u, which still exceeds "
+						"maximum size %u. This can cause compression of chunks to fail.",
+						17652,
+						9960,
+						8164)));
+#else
+		ereport(WARNING,
+				(errmsg("Estimated row size of compressed hypertable is %u. Assuming data gets "
+						"compressed to half the original size which is %u, which still exceeds "
+						"maximum size %zu. This can cause compression of chunks to fail.",
+						row_size,
+						(row_size / 2),
+						MaxHeapTupleSize)));
```

Comment:
The error message is usually smaller and without too many details, while additional details and suggestions for how to solve the problem can be added using `errdetail` or `errhint` respectively. See [Error Message Style Guide](https://www.postgresql.org/docs/current/error-style-guide.html) in the PostgreSQL manual.

```suggestion
		ereport(WARNING,
				(errmsg("compressed row size might exceed maximum row size"),
				 errdetail("Estimated row size of compressed hypertable is %u. Assuming data gets "
						"compressed to half the original size which is %u, this can exceed the "
						"maximum size of %zu and can cause compression of chunks to fail.",
						row_size,
						(row_size / 2),
						MaxHeapTupleSize)));
```